pages/cookbook/jettons.mdx

```diff
@@ -49,30 +100,41 @@ fun calculateJettonWalletAddress(ownerAddress: Address, jettonMasterAddress: Add
     return contractAddress(StateInit{code: jettonWalletCode, data: initData.toCell()});
 }
 
-contract Sample {
-    jettonWalletCode: Cell;
-    jettonMasterAddress: Address;
+message(0x7362d09c) JettonTransferNotification {
+    queryId: Int as uint64;
+    amount: Int as coins;
+    sender: Address;
+    forwardPayload: Slice as remaining;
+}
+
+contract Example with Deployable {
+    myJettonWalletAddress: Address;
+    myJettonAmount: Int as coins = 0;
 
     init(jettonWalletCode: Cell, jettonMasterAddress: Address) {
-        self.jettonWalletCode = jettonWalletCode;
-        self.jettonMasterAddress = jettonMasterAddress;
+        self.myJettonWalletAddress = calculateJettonWalletAddress(myAddress(), jettonMasterAddress, jettonWalletCode);
```

Comment:
as I already said before in previous reviews, in most cases the solution for verification is to calculate that address offchain and then just add it into the stateinit of the contract itself.

calculating the address on-chain will only work if you're 100% sure in the jetton code and data structure, or if you take them as variable parameters in the stateinit